spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/lock/JdbcLockRegistry.java

```diff
@@ -0,0 +1,232 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.jdbc.lock;
+
+import java.io.UnsupportedEncodingException;
+import java.math.BigInteger;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+import org.springframework.integration.support.locks.DefaultLockRegistry;
+import org.springframework.integration.support.locks.ExpirableLockRegistry;
+import org.springframework.integration.support.locks.LockRegistry;
+import org.springframework.messaging.MessagingException;
+import org.springframework.util.Assert;
+
+/**
+ *
+ * A {@link LockRegistry} using a shared database to co-ordinate the locks. Provides the
+ * same semantics as the {@link DefaultLockRegistry}, but the locks taken will be global,
+ * as long as the underlying database supports the "serializable" isolation level in its
+ * transactions.
+ *
+ * @author Dave Syer
+ *
+ */
+public class JdbcLockRegistry implements ExpirableLockRegistry {
+
+	private final Map<String, JdbcLock> locks = new HashMap<String, JdbcLock>();
+
+	private JdbcClient client;
+
+	public JdbcLockRegistry(JdbcClient client) {
+		this.client = client;
+	}
+
+	@Override
+	public Lock obtain(Object lockKey) {
+		Assert.isInstanceOf(String.class, lockKey);
+		String path = pathFor((String) lockKey);
+		JdbcLock lock = this.locks.get(path);
+		if (lock == null) {
+			synchronized (this.locks) {
+				lock = this.locks.get(path);
+				if (lock == null) {
+					lock = new JdbcLock(this.client, path);
+					this.locks.put(path, lock);
+				}
+			}
+		}
+		return lock;
+	}
+
+	private String pathFor(String lockKey) {
+		MessageDigest digest;
+		try {
+			digest = MessageDigest.getInstance("MD5");
+		}
+		catch (NoSuchAlgorithmException e) {
+			throw new IllegalStateException("MD5 algorithm not available.  Fatal (should be in the JDK).");
+		}
+		try {
+			byte[] bytes = digest.digest(lockKey.getBytes("UTF-8"));
+			return String.format("%032x", new BigInteger(1, bytes));
+		}
+		catch (UnsupportedEncodingException e) {
+			throw new IllegalStateException("UTF-8 encoding not available.  Fatal (should be in the JDK).");
+		}
```

Comment:
`JdbcMessageStore` uses `UUIDConverter` for this purpose:
```
private String getKey(Object input) {
	return input == null ? null : UUIDConverter.getUUID(input).toString();
}
```

Any reason do not follow with the same approach ?